dotnet/src/Connectors/Connectors.Ollama/OllamaMetadata.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Runtime.CompilerServices;
+using OllamaSharp.Models;
+using OllamaSharp.Models.Chat;
+
+namespace Microsoft.SemanticKernel.Connectors.Ollama;
+
+/// <summary>
+/// Represents the metadata of the Ollama response.
+/// </summary>
+public sealed class OllamaMetadata : ReadOnlyDictionary<string, object?>
+{
+    internal OllamaMetadata(GenerateCompletionResponseStream? ollamaResponse) : base(new Dictionary<string, object?>())
+    {
+        if (ollamaResponse is null)
+        {
+            return;
+        }
+
+        this.CreatedAt = ollamaResponse.CreatedAt;
+        this.Done = ollamaResponse.Done;
+
+        if (ollamaResponse is GenerateCompletionDoneResponseStream doneResponse)
+        {
+            this.TotalDuration = doneResponse.TotalDuration;
+            this.EvalCount = doneResponse.EvalCount;
+            this.EvalDuration = doneResponse.EvalDuration;
+            this.LoadDuration = doneResponse.LoadDuration;
+            this.PromptEvalCount = doneResponse.PromptEvalCount;
+            this.PromptEvalDuration = doneResponse.PromptEvalDuration;
+        }
+    }
+
+    internal OllamaMetadata(ChatResponseStream? message) : base(new Dictionary<string, object?>())
+    {
+        if (message is null)
+        {
+            return;
+        }
+        this.CreatedAt = message?.CreatedAt;
+        this.Done = message?.Done;
+
+        if (message is ChatDoneResponseStream doneMessage)
+        {
+            this.TotalDuration = doneMessage.TotalDuration;
+            this.EvalCount = doneMessage.EvalCount;
+            this.EvalDuration = doneMessage.EvalDuration;
+            this.LoadDuration = doneMessage.LoadDuration;
+            this.PromptEvalCount = doneMessage.PromptEvalCount;
+            this.PromptEvalDuration = doneMessage.PromptEvalDuration;
+        }
+    }
+
+    internal OllamaMetadata(ChatResponse response) : base(new Dictionary<string, object?>())
+    {
+        this.TotalDuration = response.TotalDuration;
+        this.EvalCount = response.EvalCount;
+        this.EvalDuration = response.EvalDuration;
+        this.CreatedAt = response.CreatedAt;
+        this.LoadDuration = response.LoadDuration;
+        this.PromptEvalDuration = response.PromptEvalDuration;
+        this.CreatedAt = response.CreatedAt;
+    }
+
+    /// <summary>
+    /// Time spent in nanoseconds evaluating the prompt
+    /// </summary>
+    public long PromptEvalDuration
+    {
+        get => this.GetValueFromDictionary() as long? ?? 0;
+        internal init => this.SetValueInDictionary(value);
+    }
+
+    /// <summary>
+    /// Number of tokens in the prompt
+    /// </summary>
+    public int PromptEvalCount
```

Comment:
It seems this new approach, with a property per metadata piece, deviates from the current approach other SK connectors follow - exposing metadata as a dictionary. Are there any plans to adopt the new approach for the other connectors as well, so SK has a consistent way of dealing with metadata?